testapi.pm

```diff
@@ -1457,11 +1457,9 @@ sub type_string {
     else {
         %args = @_;
     }
-    my $log = $args{secret} ? 'SECRET STRING' : $string;
     $string .= "\n" if $args{lf};
 
     if (is_serial_terminal) {
-        bmwqemu::log_call(text => $log, %args);
```

Comment:
> It's already not so nice to have that "if" in the line above in the testapi at all. Maybe we can find a better solution to not need to have console specific implementations behaving that different. 

I would say it other way around , after all in any case there is a call to `query_isotovideo('backend_type_string', {text => $piece, max_interval => $max_interval});` but in case of serial **just this call** while for screen-based cases we need to check and do dozen of other things . so better to align screen-based logic with serial terminal which is most simple one

Reply:
I mean instead of having  `if(is_serial_terminal)` in middle of function do `if(! is_serial_terminal)` and do all screen based magic and than at the end have unconditional call for `query_isotovideo('backend_type_string', {text => $piece, max_interval => $max_interval});` because it is really what happens in ALL cases just currently we have this line duplicated in several places in the code ...  But this is would be major function refactoring which goes far beyond goals which this PR trying to reach ...

Reply:
@asmorodskyi i saw your comment in the morning but i was really into some other stuff. But i dipped into the code and i had an idea, which i was planning to started tomorrow. let me provide a PR and compare later. i think it should be simple and really nice but i do not know where it can hit some side effects, will they will need extra caring

Reply:
@asmorodskyi @okurz https://github.com/os-autoinst/os-autoinst/pull/2002